### ISSUE TITLE
Load the model from a specific check point which generated minimum loss

### ISSUE THREAD

**Issue by ShrikanthSingh** [google-research/bert#970]
I am running the BERT model using tfhub similar to the example shared in the repository. I find that training loss is minimum during the 400th step and after that it is slightly higher when the  last check point is created. For the validation set the model is loaded from the last created check point i.e 636th step but i want to manually load the model from 400th step so that I can have better parameters of the model and enhance my accuracy. 
In order to do so what are the changes i should make to the functions. 

This is the loss during the last check point creation at the step 636
![image](https://user-images.githubusercontent.com/41674900/71492567-7f76e780-2838-11ea-94c5-b7590e56b423.png)

This is loss during the checkpoint creation at the step 400:
![image](https://user-images.githubusercontent.com/41674900/71492597-bd740b80-2838-11ea-9bbc-7333c94ea88a.png)

So while running the model on the validation set I want to load the model from the checkpoint 400th step and not from the 500th step. 

Kindly help me out as this can enhance the performance of my model by atleast 3-5%.

**Comment by Nidhi-kumari** [comment]
were you able to solve this? if yes can you please tell me

**Comment by ShrikanthSingh** [comment]
`prediction = estimator.predict(predict_input_fn, checkpoint_path='/content/drive/My Drive/BERT checkpoints/model.ckpt-7097')`

You can manually give the checkpoint you want to load from in the estimator.predict. Hope this helps.

**Comment by Nidhi-kumari** [comment]
can you also please tell me is there a way to store the best model based on eval loss as this code is storing based on checkpoints

**Comment by ShrikanthSingh** [comment]
> can you also please tell me is there a way to store the best model based on eval loss as this code is storing based on checkpoints

Even I was trying for that but could not find a solution as I am a begginer in TF, but on surfing over the repos on kaggle I found one implementation using Keras and might be useful for you.
[https://www.kaggle.com/xhlulu/disaster-nlp-keras-bert-using-tfhub](url)

Since it uses Functional API you can add callbacks for saving the best model easily.
